downloader/src/main/kotlin/vcs/GitWorkingTree.kt

```diff
@@ -39,6 +41,18 @@ open class GitWorkingTree(workingDir: File, private val gitBase: GitBase) : Work
         return File(dotGitDir, "shallow").isFile
     }
 
+    override fun getNested(): Map<String, VcsInfo> {
+        val root = getRootPath()
+
+        val paths = gitBase.run(root, "submodule", "--quiet", "foreach", "--recursive", "echo \$sm_path").stdout.lines()
+                .filter { it.isNotBlank() }
+
+        return paths.associateWith { path ->
+            val workingTree = Git().getWorkingTree(root.resolve(path))
```

Comment:
It just occurred to me we could also write this directly as `GitWorkingTree(root.resolve(path), gitBase).getInfo()`.

Reply:
Done.